packages/@uppy/transloadit/src/index.ts

```diff
@@ -144,67 +145,21 @@ interface BaseOptions extends PluginOpts {
   limit?: number
   clientName?: string | null
   retryDelays?: number[]
+  assemblyOptions?:
+    | AssemblyOptions
+    | ((
+        file?: UppyFile<M, B> | null,
+        options?: TransloaditOptions<M, B>,
+      ) => Promise<AssemblyOptions> | AssemblyOptions)
 }
 
-export type TransloaditOptions<M extends Meta, B extends Body> = BaseOptions &
-  (
-    | {
-        assemblyOptions?:
-          | AssemblyOptions
-          | ((
-              file?: UppyFile<M, B> | null,
-              options?: AssemblyOptions,
-            ) => Promise<AssemblyOptions> | AssemblyOptions)
-        /** @deprecated use `assemblyOptions` instead */
-        getAssemblyOptions?: never | null
-        /** @deprecated use `assemblyOptions` instead */
-        params?: never | null
-        /** @deprecated use `assemblyOptions` instead */
-        fields?: never | null
-        /** @deprecated use `assemblyOptions` instead */
-        signature?: never | null
-      }
-    | {
-        /** @deprecated use `assemblyOptions` instead */
-        getAssemblyOptions?: (
-          file?: UppyFile<M, B> | null,
-        ) => AssemblyOptions | Promise<AssemblyOptions>
-        assemblyOptions?: never
-        /** @deprecated use `assemblyOptions` instead */
-        params?: never | null
-        /** @deprecated use `assemblyOptions` instead */
-        fields?: never | null
-        /** @deprecated use `assemblyOptions` instead */
-        signature?: never | null
-      }
-    | {
-        /** @deprecated use `assemblyOptions` instead */
-        params?: AssemblyParameters | null
-        /** @deprecated use `assemblyOptions` instead */
-        fields?: { [name: string]: number | string } | string[] | null
-        /** @deprecated use `assemblyOptions` instead */
-        signature?: string | null
-        /** @deprecated use `assemblyOptions` instead */
-        getAssemblyOptions?: never | null
-        assemblyOptions?: never
-      }
-  )
-
```

Comment:
While we are breaking, can we rename service to endpoint as to be consistent with other integrations?

Reply:
I can do it in a separate PR 👍